src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/WriteableBitmap.cs

```diff
@@ -1012,20 +1017,27 @@ bool      backwardsCompat
 
                     Lock();
 
-                    MILUtilities.MILCopyPixelBuffer(
-                        pDest,
-                        outputBufferSize,
-                        (uint) _backBufferStride.Value,
-                        destBufferBitOffset,
-                        pSource,
-                        inputBufferSize,
-                        (uint) sourceBufferStride,
-                        sourceBufferBitOffset,
-                        (uint) sourceRect.Height,
-                        copyWidthInBits);
-
-                    AddDirtyRect(destinationRect);
-                    Unlock();
+                    try
+                    {
+                        MILUtilities.MILCopyPixelBuffer(
+                            pDest,
+                            outputBufferSize,
+                            (uint) _backBufferStride.Value,
+                            destBufferBitOffset,
+                            pSource,
+                            inputBufferSize,
+                            (uint) sourceBufferStride,
+                            sourceBufferBitOffset,
+                            (uint) sourceRect.Height,
+                            copyWidthInBits);
+                        AddDirtyRect(destinationRect);
+                    }
+                    finally
+                    {
+                        // The MILUtilities.MILCopyPixelBuffer may throw ArgumentException. We put Unlock method within the finally block to ensure that the Lock and Unlock methods are called in pairs
```

Comment:
OK then I would mention that and remove the pair comment which is unnecessary and technically might not even be true

```suggestion
                        // MILUtilities.MILCopyPixelBuffer may throw ArgumentException (e.g. for invalid stride)
```

Not sure the issue reference is that important in this case, the comment is self-explanatory.